lib/mongo/crypt/encryption_io.rb

```diff
@@ -51,7 +54,37 @@ def find_keys(filter)
       #
       # @return [ Mongo::Operation::Insert::Result ] The insertion result
       def insert(document)
-        @collection.insert_one(document)
+        @key_vault_collection.insert_one(document)
+      end
+
+      # Get collection info for a collection matching the provided filter
+      #
+      # @param [ Hash ] filter
+      #
+      # @return [ Hash ] The collection information
+      def collection_info(filter)
+        result = @client.database.list_collections
+
+        name = filter['name']
+        result.find { |r| r['name'] == name }
```

Comment:
@p-mongo I was trying to do something like Python's list_collections(filter) method -- https://api.mongodb.com/python/current/api/pymongo/database.html#pymongo.database.Database.list_collections. What do you think is the best approach?

Reply:
The server command supports a filter argument (https://docs.mongodb.com/manual/reference/command/listCollections/). Our helper doesn't apparently (https://api.mongodb.com/ruby/current/Mongo/Database.html#collections-instance_method). Amending the helper to pass the filter seems like a valid strategy, but if the inline filtering here works it seems fine to do this as well.